engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala

```diff
@@ -355,34 +359,36 @@ case class WorkflowExecutionActor(workflowDescriptor: EngineWorkflowDescriptor,
     if (runnableCalls.nonEmpty) workflowLogger.info("Starting calls: " + runnableCalls.mkString(", "))
 
     // Each process returns a Try[WorkflowExecutionDiff], which, upon success, contains potential changes to be made to the execution store.
-    val executionDiffs = runnableScopes map {
-      case k: CallKey if isInBypassedScope(k, data) => processBypassedScope(k, data)
-      case k: DeclarationKey if isInBypassedScope(k, data) => processBypassedScope(k, data)
-      case k: BackendJobDescriptorKey => processRunnableJob(k, data)
-      case k: ScatterKey => processRunnableScatter(k, data, isInBypassedScope(k, data))
-      case k: ConditionalKey => processRunnableConditional(k, data)
-      case k: CollectorKey => processRunnableCollector(k, data, isInBypassedScope(k, data))
-      case k: SubWorkflowKey => processRunnableSubWorkflow(k, data)
-      case k: StaticDeclarationKey => processRunnableStaticDeclaration(k)
-      case k: DynamicDeclarationKey => processRunnableDynamicDeclaration(k, data)
-      case k =>
-        val exception = new UnsupportedOperationException(s"Unknown entry in execution store: ${k.tag}")
-        self ! JobInitializationFailed(k, exception)
-        Failure(exception)
+    val diffs = runnableScopes map { scope =>
+      scope -> Try(scope match {
+        case k: CallKey if isInBypassedScope(k, data) => processBypassedScope(k, data)
+        case k: DeclarationKey if isInBypassedScope(k, data) => processBypassedScope(k, data)
+        case k: BackendJobDescriptorKey => processRunnableJob(k, data)
+        case k: ScatterKey => processRunnableScatter(k, data, isInBypassedScope(k, data))
+        case k: ConditionalKey => processRunnableConditional(k, data)
+        case k: CollectorKey => processRunnableCollector(k, data, isInBypassedScope(k, data))
+        case k: SubWorkflowKey => processRunnableSubWorkflow(k, data)
+        case k: StaticDeclarationKey => processRunnableStaticDeclaration(k)
+        case k: DynamicDeclarationKey => processRunnableDynamicDeclaration(k, data)
+        case k => Failure(new UnsupportedOperationException(s"Unknown entry in execution store: ${k.tag}"))
+      }).flatten
+    } map {
+      case (_, Success(value)) => Success(value)
+      case (scope, Failure(throwable)) =>
+        self ! JobInitializationFailed(scope, throwable)
+        Failure(throwable)
+    } collect {
+      case Success(value) => value
```

Comment:
TOL: Doesn't that mean that we're potentially going to start new jobs even though the workflow is doomed to fail if one of the `JobInitializationFailed` message was sent ?